s3torchconnector/src/s3torchconnector/s3map_dataset.py

```diff
@@ -32,19 +33,17 @@ def __init__(
         self._transform = transform
         self._region = region
         self._client = None
-        self._dataset_object_store = None
+        self._bucket_key_pairs = None
 
     @property
     def region(self):
         return self._region
 
     @property
-    def _dataset_objects(self) -> List[S3Reader]:
-        if self._dataset_object_store is None:
-            self._dataset_object_store = list(
-                self._get_dataset_objects(self._get_client())
-            )
-        return self._dataset_object_store
+    def _dataset_bucket_key_pairs(self) -> List[S3BucketKey]:
+        if self._bucket_key_pairs is None:
+            self._bucket_key_pairs = list(self._get_dataset_objects(self._get_client()))
```

Comment:
We've changed the return type of `self._get_dataset_objects`, so also need to change the type in the definition of `__init__`